guides/effectsize_exemplar_within.Rmd

```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
 
 For those accustomed to Cohen's *d*, it's important to be aware that $\eta_{G}^{2}$ is typically smaller, with a Cohen's d of 0.2 being equivalent to a $\eta_{G}^{2}$ of around 0.02. Also, the actual number has little meaning beyond its scale relative to other effects. 
 
 ```{r within-anova}
 results = afex::aov_ez(
   data = data, 
-  id = 'id', # subject id column
+  id = 'subject', # subject id column
   dv = 'response_time', # dependent variable
   within = c('layout', 'size', 'color'), # within-subject independent variables
   between = NULL ,# between-subject independent variables
+  fun_aggregate = mean, # average multiple repetitions together for each subject*condition
   anova_table = list(es = 'ges') # effect size = generalized eta squared
 )
 ```
 
-*Note: the warning indicates that the aov_ez function automatically collapses repetitions into a mean, which may be a problem if an experiment is not fully counterbalanced. This example, however, has every subject running in every combination of conditions, so simple collapsing is the correct procedure.*
+*Note: `fun_aggregate = mean` collapses repetitions into a mean, which may be a problem if an experiment is not fully counterbalanced. This example, however, has every subject running in every combination of conditions, so simple collapsing is the correct procedure.*
 
 ```{r  within-anova-cleanup}
 anovaResults <- 
   results$anova_table %>% 
   rownames_to_column('effect') %>%  # put effect names in a column
   select(-`Pr(>F)`) # no need to show p-values
   
-anovaResults %>% 
-  tidy() %>% 
-  rename(F = statistic)
+anovaResults %>% as.tibble()
+```
+
+Looking at the `F` and `ges` (generalized eta squared) columns, there appear to be strong main effects for `layout` and `size` and an interaction between `layout` and `size`. However `color` and the other 2-way and 3-way interactions show only negligeable effects.
+
+
+
+### Bootstrapping
+
+<mark> Draft. Needs work. </mark>
+
+But that only gives us one value, whereas we want a confidence interval.
+
+We'll use a technique called bootstrapping, which checks the effect size for many random permutations of the data.
+
+1. Randomly sample with replacement (meaning the same value might be sampled more than once) to build a new dataset
+1. Perform the analysis on this new dataset
+1. Do that many times
+1. Sort the results for each effect and find the 95% confidence interval
+
+
```

Comment:
"Bootstrapping works because of the [central limit theorem](https://en.wikipedia.org/wiki/Central_limit_theorem). This allows us to get calculate an interval estimate (95% confidence interval of `ges`) rather than a point estimate (`ges`)

Reply:
I'd rather not rely on a reader's understanding of the CLM. I remember the concept of bootstrapping taking me a while to grasp in grad school. Here's a first pass at a simple explanation of bootstrapping:

We'll use a technique called bootstrapping, which checks the effect size for many randomized samples of the data. Importantly, bootstrapping samples "with replacement", meaning that items can be sampled more than once or not at all. If a small subset of the observations are driving an effect, they won't impact all of the samples. Consequently, the spread of the bootstrapped confidence intervals shows how consistent the results are for different samples.

Reply:
Here's a short explanation I used: "Briefly, it consists of generating many alternative datasets from the experimental data by randomly drawing observations with replacement. The variability across these datasets is assumed to approximate sampling error and is used to compute so- called bootstrap confidence intervals."

Reply:
I think the key is that (in most cases) the distribution of bootstrap samples approximates the sampling distribution.